### PR TITLE
hide org info from requests that don't need it

### PR DESCRIPTION
**Pull request by camhashemi**
@andrewconner (@ryanpbrewster)

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserProfileController.scala [hunk 1]
```diff
@@ -62,11 +62,16 @@ class UserProfileController @Inject() (
         val numConnections = userConnectionRepo.getConnectionCount(profile.userId)
         val numFollowers = userProfileCommander.countFollowers(profile.userId, viewer.map(_.id.get))
         val userBio = userValueRepo.getValueStringOpt(profile.userId, UserValueName.USER_DESCRIPTION)
-        val orgMemberships = orgMembershipRepo.getAllByUserId(profile.userId)
-        val pendingOrgs = orgInviteRepo.getByInviteeIdAndDecision(profile.userId, InvitationDecision.PENDING).groupBy(_.organizationId).keys.map { orgId =>
-          organizationInfoCommander.getOrganizationInfo(orgId, viewer.flatMap(_.id))
+        val (orgInfos, pendingOrgs) = {
+          if (viewer.flatMap(_.id).contains(profile.userId)) {
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserProfileController.scala]: `viewer.map(_.id.get)`

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserProfileController.scala [hunk 2]
```diff
@@ -62,11 +62,16 @@ class UserProfileController @Inject() (
...
+            val orgMemberships = orgMembershipRepo.getAllByUserId(profile.userId)
+            val orgs = orgMemberships.map { orgMembership => organizationInfoCommander.getOrganizationInfo(orgMembership.organizationId, viewer.flatMap(_.id)) }
+            val pendingOrgs = orgInviteRepo.getByInviteeIdAndDecision(profile.userId, InvitationDecision.PENDING).groupBy(_.organizationId).keys.map { orgId =>
+              organizationInfoCommander.getOrganizationInfo(orgId, viewer.flatMap(_.id))
+            }
+            (orgs, pendingOrgs)
+          } else (Seq.empty, Set.empty)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserProfileController.scala]: Maybe this whole thing could be

```
viewer.map(_.id.get).filter(_ == profile.userId).map { userId =>
   ...
}.getOrElse(Seq.empty, Set.empty)
```

Then you could avoid mixing/matching `profile.userId` and `viewer.map(_.id.get)`

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/UserProfileController.scala]: that's very cool, thank you

**Comment by ryanpbrewster** [comment]
Beautiful.

**Pull request merged**